### ISSUE TITLE
Sort alignments by base incorrect for lower case sequence. (was Sort alignments by base shows most abundant base in hg38)

### ISSUE THREAD

**Issue by kimin0402** [igvteam/igv#929]
Hi, I think it's a very simple issue, but I couldn't find the answer for it.

I'm trying to analyze a bam file on igv, and when I sort alignments by base on an alignment track, somehow the most abundant base or the reference base shows on top of the track. 

This doesn't happen when I'm using hg19 as a reference. 

Why is this happening?

**Comment by jrobinso** [comment]
I don't understand what you mean,  maybe you could post some screenshots or better a test case.   You can't use both hg38 and hg19 as a reference for the same BAM,  you will definitely get different sorting behavior,  you should use the reference the BAM was aligned against.

I think I'll need a test case to understand your issue,  but what do you expect sorting by base to do?

**Comment by kimin0402** [comment]
Okay, sorry if my question was misleading.

When I use 'sort alignments by base' at a certain position, I am usually looking at a SNV site, so the expected behavior is to see reads with ALT bases at the sorting position come at the top of the alignment track and reads with REF bases are placed below the reads with ALT bases.  

When using GRCh37 as a reference and seeing GRCh37 aligned bam, this works fine.

but somehow, when using hg38 as a reference and seeing hg38 aligned bam, the behavior is completely other way around. The ALT bases are sorted at the bottom and REF bases (usually the most abundant ones) are sorted at the top. 

If this explanation is not clear, I'll try to post a screenshot. 

My lab mate is facing a same issue. The reference sequence track at the bottom of IGV shows colored upper case letters for 37 and lower case letters for 38. Could this be a reason?

My igv version is 2.7.2

Thank you in advance.

**Comment by jrobinso** [comment]
I can't think of any reason for the behavior you describe, there is nothing hardcoded about hg39 vs hg19.   Release 2.7.2 was quite some time ago,  I don't recall of course what issues might have been fixed in the meantime, but I just checked with release 2.9.2 and it works as expected.   Keep in mind that IGV sorts based on the position and the center of the view, but I'm sure you know that.

**Comment by kimin0402** [comment]
Thank you @jrobinso I think you understand my question, but for anyone sharing a similar problem or anyone interested in solving this issue I'd like to post screenshots of my IGV. By the way, I updated it to the most recent version (2.9.2) IGV for linux with java included. 

Sort alignment by base in GRCh37 (ctrl + s)
![37](https://user-images.githubusercontent.com/50519841/110738129-17ac9100-8272-11eb-9645-d0691fa81b08.JPG)

Sort alignment by base in GRCh38 (ctrl + s)
![38](https://user-images.githubusercontent.com/50519841/110738134-1a0eeb00-8272-11eb-9414-befd165c8856.JPG)

**Comment by jrobinso** [comment]
@kimin0402 Sorry but what is "ctrl-S"?   Is this some magic key I forgot.   I sort by right-clicking and selecting from the menu.

**Comment by jrobinso** [comment]
Hmmm,  I wonder if its the lower cased sequence, being improperly considered a mismatch.   Can you try it in a region that is upper case (not in a repeat region)?

**Comment by kimin0402** [comment]
@jrobinso Thank you!

It was due to the lower case/upper case difference. I guess I have to take that fact into account when looking at a repeat region. Thank you for a quick reply to a very simple question. Really appreciate your kindness.

By the way, ctrl - s is a shortkey for sorting alignments. It repeats the most recent sort. My lab members use it frequently and I found the reference at this [link](http://software.broadinstitute.org/software/igv/book/export/html/6).
